inference-engine/tests/functional/plugin/shared/src/single_layer_tests/concat_4D.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <memory>
+#include <functional>
+
+#include "ie_core.hpp"
+
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "functional_test_utils/precision_utils.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+
+#include "single_layer_tests/concat_4D.hpp"
+
+namespace LayerTestsDefinitions {
+
+    std::string Concat4DLayerTest::getTestCaseName(const testing::TestParamInfo<concat4DParamsTuple> &obj) {
+        int axis;
+        std::vector<size_t> inputShapes;
+        InferenceEngine::Precision netPrecision;
+        InferenceEngine::Precision inPrc, outPrc;
+        InferenceEngine::Layout inLayout, outLayout;
+        std::string targetName;
+        std::map<std::string, std::string> config;
+        std::tie(inputShapes, netPrecision, targetName, config) = obj.param;
+        std::ostringstream result;
+        result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+        result << "netPRC=" << netPrecision.name() << "_";
+        result << "trgDev=" << targetName << "_";
+        return result.str();
+    }
+
+    void Concat4DLayerTest::SetUp() {
+        int axis = 1;
+        InferenceEngine::SizeVector inputShape;
+        InferenceEngine::Precision netPrecision;
+        std::map<std::string, std::string> additional_config;
+        std::tie(inputShape, netPrecision, targetDevice, additional_config) = this->GetParam();
+        configuration.insert(additional_config.begin(),additional_config.end());
+
+        const int seed = 0;
+        std::mt19937 gen(static_cast<float>(seed));
+
+        auto generateFloatNumbers = [gen](std::size_t vec_len, float min, float max) mutable {
```

Comment:
Are there any test utils where this function could be placed? It seems like it is used in many tests

Reply:
I indeed use it a lot, since testing infrastructure does not provide reliable way to generate numbers, there was quite a resistance when i wanted to do it myself last time though, I think I can make GNA specific header for it though